feat!: enable payments with predicates (#815)
FuelLabs/fuels-rs: docs/src/predicates/send-spend-predicate.md

{{#include ../../../examples/predicates/src/lib.rs:predicate_coins}}
```

> Now we can use the predicate abigen, which will create a predicate encoder instance for us. To spend the funds that are now locked in the predicate, we have to provide two out of three signatures whose public keys match the ones we defined in the predicate. In this example, the signatures are generated from an array of zeros.

Reviewer: Shortening it to improve a bit of clarity.

```suggestion
Now we can use the predicate abigen to create a predicate encoder instance for us. To spend the funds now locked in the predicate, we must provide two out of three signatures whose public keys match the ones we defined in the predicate. In this example, the signatures are generated from an array of zeros.
```

```rust,ignore
{{#include ../../../examples/predicates/src/lib.rs:predicate_load}}
```

Next, we transfer some assets from a wallet to the created predicate. We also confirm that the funds are indeed transferred.

```rust,ignore
{{#include ../../../examples/predicates/src/lib.rs:predicate_receive}}
```

> We can use the `transfer` method from the [Account](../getting-started/account.md) trait to transfer the assets. If the provided predicate data is correct the `receiver` wallet will get the funds, and we will verify that the amount is indeed correct.

Reviewer: ```suggestion
We can use the `transfer` method from the [Account](../getting-started/account.md) trait to transfer the assets. If the predicate data is correct, the `receiver` wallet will get the funds, and we will verify that the amount is correct.
```
